hal/src/sercom/v2/uart/reg.rs

```diff
@@ -0,0 +1,411 @@
+//! Register-level access to UART configuration
+
+use super::{BaudMode, BitOrder, CharSizeEnum, Parity, StopBits};
+
+use crate::sercom::v2::*;
+use crate::target_device as pac;
+
+#[cfg(any(feature = "samd11", feature = "samd21"))]
+use pac::sercom0::usart::ctrla::MODE_A;
+
+#[cfg(feature = "min-samd51g")]
+use pac::sercom0::usart_int::ctrla::MODE_A;
+
+use crate::time::Hertz;
+
+pub struct Registers<S: Sercom> {
+    sercom: S,
+}
+
+// SAFETY: It is safe to implement Sync for Registers, because it erases the
+// interior mutability of the PAC SERCOM struct.
+unsafe impl<S: Sercom> Sync for Registers<S> {}
+
+impl<S: Sercom> Registers<S> {
+    /// Create a new `Registers` instance
+    #[inline]
+    pub(super) fn new(sercom: S) -> Self {
+        Self { sercom }
+    }
+
+    /// Helper function to access the underlying `USART` from the given `SERCOM`
+    #[cfg(any(feature = "samd11", feature = "samd21"))]
+    #[inline]
+    fn usart(&self) -> &pac::sercom0::USART {
+        self.sercom.usart()
+    }
+
+    /// Helper function to access the underlying `USART_INT` from the given
+    /// `SERCOM`
+    #[cfg(feature = "min-samd51g")]
+    #[inline]
+    fn usart(&self) -> &pac::sercom0::USART_INT {
+        self.sercom.usart_int()
+    }
+
+    #[cfg(feature = "dma")]
+    /// Get a pointer to the `DATA` register
+    pub(super) fn data_ptr<T>(&self) -> *mut T {
+        self.usart().data.as_ptr() as *mut _
+    }
+
+    /// Free the `Registers` struct and return the underlying `Sercom` instance
+    #[inline]
+    pub(super) fn free(self) -> S {
+        self.sercom
+    }
+
+    /// Reset the SERCOM peripheral
+    #[inline]
+    pub(super) fn swrst(&mut self) {
+        self.usart().ctrla.write(|w| w.swrst().set_bit());
+        while self.usart().syncbusy.read().swrst().bit_is_set() {}
+    }
+
+    /// Configure the SERCOM to use internal clock mode
+    #[inline]
+    pub(super) fn configure_mode(&mut self) {
+        self.usart()
+            .ctrla
+            .modify(|_, w| w.mode().variant(MODE_A::USART_INT_CLK));
+    }
+
+    /// Configure the `SERCOM`'s Pads according to RXPO and TXPO
+    #[inline]
+    pub(super) fn configure_pads(&mut self, rxpo: u8, txpo: u8) {
+        self.usart().ctrla.modify(|_, w| unsafe {
+            w.rxpo().bits(rxpo);
+            w.txpo().bits(txpo)
+        });
+    }
+
+    /// Configure the character size
+    #[inline]
+    pub(super) fn set_char_size(&mut self, bits: u8) {
```

Comment:
The other functions take the corresponding `enum`, but not this one. The changes I proposed in the `char_size` module should help that.